fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/provider/impl/S3Provider.java

```diff
@@ -378,39 +378,36 @@ private void pushFhirJsonsToCos(InputStream in, int dataLength) throws Exception
             chunkData.setUploadId(uploadId);
         }
 
-        // Only if we're greater than zero....Otherwise there is nothing to upload.
-        if (chunkData.getBufferStream().size() > 0) {
-            chunkData.getCosDataPacks().add(BulkDataUtils.multiPartUpload(client, bucketName, itemName, uploadId, in, dataLength, chunkData.getPartNum()));
-            logger.info("pushFhirJsonsToCos: '" + dataLength + "' bytes were successfully appended to COS object - '" + itemName + "' uploadId='" + uploadId + "'");
-            chunkData.setPartNum(chunkData.getPartNum() + 1);
-            chunkData.getBufferStream().reset();
-
-            // Close it out...
-            if (chunkData.getPageNum() > chunkData.getLastPageNum() || chunkData.isFinishCurrentUpload()) {
-                BulkDataUtils.finishMultiPartUpload(client, bucketName, itemName, uploadId, chunkData.getCosDataPacks());
-                // Partition status for the exported resources, e.g, Patient[1000,1000,200]
-                if (chunkData.getResourceTypeSummary() == null) {
-                    chunkData.setResourceTypeSummary(fhirResourceType + "[" + chunkData.getCurrentUploadResourceNum());
-                    if (chunkData.getPageNum() > chunkData.getLastPageNum()) {
-                        chunkData.setResourceTypeSummary(chunkData.getResourceTypeSummary() + "]");
-                    }
-                } else {
-                    chunkData.setResourceTypeSummary(chunkData.getResourceTypeSummary() + "," + chunkData.getCurrentUploadResourceNum());
-                    if (chunkData.getPageNum() > chunkData.getLastPageNum()) {
-                        chunkData.setResourceTypeSummary(chunkData.getResourceTypeSummary() + "]");
-                    }
+        chunkData.getCosDataPacks().add(BulkDataUtils.multiPartUpload(client, bucketName, itemName, uploadId, in, dataLength, chunkData.getPartNum()));
```

Comment:
looks like a big change, but I just removed the outer if...part of the "experimental" change mentioned above

Reply:
oh, and removed the extra check on line 387 since we're no longer looping through the DTOs and re-setting the page in the middle of the writeResources

Reply:
backed these out